### PR TITLE
[DO-1641] Add pep row clickable

### PR DESCRIPTION
**Pull request by Iva-khar**
All row space is clickable but text is selectable & copyable also
image

### CODE REVIEW

File: src/components/pages/datasets/pep/PepList.jsx [hunk 1]
```diff
@@ -98,11 +99,16 @@ const PepList = () => {
           'updated_at',
         ]}
         axiosConfigs={{ useProjectToken: true }}
+        rowLinkUrl={match.url}
+        history={history}
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepList.jsx:103]: just add onRowClick event for Table component

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: src/components/pages/datasets/pep/PepList.jsx (hunk 1 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepList.jsx]: done

File: src/components/pages/datasets/pep/PepList.jsx [hunk 2]
```diff
@@ -4,8 +4,9 @@ import Table from 'components/table/Table';
 import PageBox from 'components/pages/PageBox';
 import { useTranslation } from 'react-i18next';
 import { dateFormatISO } from 'utils';
+import { ReactRouterPropTypes } from '../../../../utils/prop-types';
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepList.jsx:7]:
```suggestion
import { ReactRouterPropTypes } from 'utils/prop-types';
```

File: src/components/pages/datasets/pep/PepList.jsx [hunk 3]
```diff
@@ -98,11 +99,17 @@ const PepList = () => {
           'updated_at',
         ]}
         axiosConfigs={{ useProjectToken: true }}
+        onRowClick={(row) => {
+          history.push(`${match.url}${row.id}`);
```
Comment (roman-tiukh) [inline src/components/pages/datasets/pep/PepList.jsx:103]:
```suggestion
          history.push(`${match.url}${row.id}/`);
```

File: src/components/table/Table.jsx [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { ChevronDown, ChevronUp } from 'react-feather';
 import LoadingIcon from 'components/LoadingIcon';
 import { useTranslation } from 'react-i18next';
 import FilterField from 'components/filter-fields/FilterField';
+import { ReactRouterPropTypes } from '../../utils/prop-types';
```
Comment (roman-tiukh) [inline src/components/table/Table.jsx:10]: unused import

File: src/components/table/Table.jsx [hunk 5]
```diff
@@ -66,10 +67,19 @@ const Table = (props) => {
...
+            const selection = window.getSelection();
+            if (onRowClick && selection.type !== 'Range') {
+              onRowClick(row);
+            }
+          }}
+        >
           {columns.map((col) => (
             <td key={col.prop} className="border-b">
-              {(col.render ? col.render(row[col.prop], row) : row[col.prop]) || '---'}
+              <span className="cursor-text">{(col.render ? col.render(row[col.prop], row) : row[col.prop]) || '---'}</span>
```
Comment (roman-tiukh) [inline src/components/table/Table.jsx:82]:
```suggestion
              <span className="cursor-text">
                {(col.render ? col.render(row[col.prop], row) : row[col.prop]) || '---'}
              </span>
```

File: src/components/table/Table.jsx [hunk 6]
```diff
@@ -66,10 +67,19 @@ const Table = (props) => {
     }
     if (tc.data.length) {
       return tc.data.map((row, i) => (
-        <tr key={row.id || i}>
+        <tr
+          className={onRowClick ? 'cursor-pointer hover:bg-gray-200' : ''}
```
Comment (roman-tiukh) [inline src/components/table/Table.jsx:71]:
```suggestion
          className={onRowClick ? 'cursor-pointer hover:bg-gray-200' : undefined}
```

File: src/components/table/Table.jsx [hunk 7]
```diff
@@ -66,10 +67,19 @@ const Table = (props) => {
...
+          key={row.id || i}
+          onClick={() => {
+            const selection = window.getSelection();
+            if (onRowClick && selection.type !== 'Range') {
+              onRowClick(row);
+            }
```
Comment (roman-tiukh) [inline src/components/table/Table.jsx:77]:
```suggestion
            if (onRowClick && window.getSelection().type !== 'Range') {
              onRowClick(row);
            }
```

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: src/components/pages/datasets/pep/PepList.jsx (hunk 3 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepList.jsx:103]: done

File: src/components/pages/datasets/pep/PepList.jsx (hunk 2 above)
Comment (Iva-khar) [reply src/components/pages/datasets/pep/PepList.jsx:7]: done

File: src/components/table/Table.jsx (hunk 4 above)
Comment (Iva-khar) [reply src/components/table/Table.jsx:10]: deleted

File: src/components/table/Table.jsx (hunk 5 above)
Comment (Iva-khar) [reply src/components/table/Table.jsx:82]: done

File: src/components/table/Table.jsx (hunk 6 above)
Comment (Iva-khar) [reply src/components/table/Table.jsx:71]: done

File: src/components/table/Table.jsx (hunk 7 above)
Comment (Iva-khar) [reply src/components/table/Table.jsx:77]: done

**Review by roman-tiukh — APPROVED** [review]

**Comment by roman-tiukh** [comment]
don`t merge this PR for now

**Review by AlenaYanish — APPROVED** [review]

**Pull request merged**